File: 백준/Bronze/21756. 지우개/지우개.py
n = int(input())
num = [i for i in range(1, n+1)]

while len(num) > 1:
    # print("num1 : ", num)
    tmp = []
    for i in range(len(num)):
        if i % 2 != 0:
            tmp.append(num[i])
    
    num = tmp
    # print("num2 : ", num)
    # print("-----")

print(num[0])

File: 백준/Bronze/21756. 지우개/지우개.cc
#include <iostream>
#include <vector>

using namespace std;

int main()
{
	int n;
	cin >> n;

	vector<int> num;
	for (int i = 1; i <= n; i++)
		num.push_back(i);

	while (num.size() > 1) {
		vector<int> tmp;

		for (int i = 0; i < num.size(); i++) {
			if (i % 2 != 0) tmp.push_back(num[i]);
		}

		num = tmp;
	}

	cout << num[0];

	return 0;
}